%----------------------------------------------------------------------------------------
%	PACKAGES AND THEMES
%----------------------------------------------------------------------------------------
\documentclass[aspectratio=169,xcolor=dvipsnames]{beamer}
\usetheme{SimplePlus}

\usepackage{algorithm,algorithmic}
\usepackage{hyperref}
\usepackage{amsmath}
\usepackage{graphicx} % Allows including images
\usepackage{booktabs} % Allows the use of \toprule, \midrule and \bottomrule in tables

\renewcommand{\algorithmicrequire}{\textbf{Input:}}
\renewcommand{\algorithmicensure}{\textbf{Output:}}

%----------------------------------------------------------------------------------------
%	TITLE PAGE
%----------------------------------------------------------------------------------------

\title[short title]{A multi-objective adaptive evolutionary algorithm to extract communities in networks} % The short title appears at the bottom of every slide, the full title is only on the title page

\author[Haris-Ali] {Haris Karim Ladhani \& Ali Hamza}

\institute[HU-CI] % Your institution as it will appear on the bottom of every slide, may be shorthand to save space
{
    CS 451 - Computational Intelligence \\
    Habib University % Your institution for the title page
}
\date{\today} % Date, can be changed to a custom date


%----------------------------------------------------------------------------------------
%	PRESENTATION SLIDES
%----------------------------------------------------------------------------------------

\begin{document}

\begin{frame}
    % Print the title page as the first slide
    \titlepage
\end{frame}

\begin{frame}{Overview}
    % Throughout your presentation, if you choose to use \section{} and \subsection{} commands, these will automatically be printed on this slide as an overview of your presentation
    \tableofcontents
\end{frame}

%------------------------------------------------
\section{Introduction}
%------------------------------------------------

\begin{frame}{Introduction}
\begin{itemize}
    \item Many complex systems such as social networks, protein networks can be abstracted into networks i.e. large graphs
    \item Analysis techniques of these networks require studying various patterns within them
    \item Common patterns include recognizing substructures within the graphs such as cliques
    \item This paper analyzes a social network data set and attempts to find communities within it via a genetic algorithm to reduce the complexity of the problem
    \item Communities can be defined by various definitions therefore the paper uses two fitness functions.
    \item \textit{max(internal links) and min(external links)}
\end{itemize}

\end{frame}

\begin{frame}{Problem Description}
    The problem at hand from a CS theoretical point of view is simply a clique, or sub graph, finding problem. 
    %How to find a clique in a graph
    %Clique finding - formal algorithm
    %NP Complete
\end{frame}

\begin{frame}{Motivation}
    %Genetic algorithm part of FYP 
    %Looking for cliques in protient networks
    %Opportunity to explore non-protein network solutions
\end{frame}

%------------------------------------------------

%------------------------------------------------
\section{Genetic Algorithm Formulation}
%------------------------------------------------
\begin{frame}{Chromosome Representation}
    \begin{itemize}
        \item an array of $N$, number of nodes in the graph, genomes and each gene ranges from 1 to $N$
        \item there are 2 vertices, i and j, and they are adjacent. 
    \end{itemize}
\end{frame}

%------------------------------------------------

\begin{frame}{Selection}
    \begin{itemize}
        \item Roulette Wheel \\
        \[g_{avg} = \frac{g_{1} + g_{2} + ... + g_{N}}{}\]
        \[\sigma = \sqrt{\frac{1}{n}(\sum_{i=1}^{N}(g_{i} - g_{avg})^{2})\]
        \[\omega = \frac{g_{avg} + 1}{\delta}\]
    \end{itemize}
\end{frame}

\begin{frame}{Fitness Function}
   
\end{frame}


\begin{frame}{Adaptive Crossover and Mutation}
The crossover function is adaptive, based on various parameters and occurs based on various circumstances such: 
\begin{itemize}
    \item Individuals with poor fitness are more likely to crossover
    \item Individuals with high fitness are less likely to crossover near the termination of algorithm
    \item Individuals with high fitness but are more likely to mutate near the termination of algorithm
    \item Populations with low deviation in fitness are less likely to crossover and more likely to mutate
    \item Populations with high deviation in fitness are more likely to crossover and less likely to mutate 
\end{itemize}
\newpage
The probabilities of these two events can be defined as follows:
$$p_c = 0.5 \times \frac{1}{1+e^{\frac{-k_1}{\omega}}} + 0.4$$
$$p_m = \frac{k_2}{5\times(1+3^{\frac{1}{\omega}})}$$

where $k_1 \in (1, \infty), k_2 \in (0,1)$

\end{frame}

%------------------------------------------------

\begin{frame}{Method of Crossover \& Mutation}
   
\end{frame}


\begin{frame}{Algorithm}
    \begin{algorithm}[H]
        \begin{algorithmic}[1]
        \STATE \textbf{Initialization:} Adaptive parameters: adaptive crossover probability $p_{c}$ of
        population $P$, adaptive mutation probability $p_{m}$ of population $P$
        \STATE $P \leftarrow Initialization(Population);$
        \STATE While $Termination(Generation)$;
        \STATE $P_{parent} \leftarrow Select(P)$
        \STATE $p_{c}, p_{m} \leftarrow Adaptive();$
        \STATE $p_{cross} \leftarrow Crossover(P_{parent}, p_{c})$
        \STATE $p_{cross} \leftarrow Mutation(P_{cross}, p_{m})$
        \STATE $P \leftarrow Update(P_{child})$
        \STATE $ElitePool();//$ {Update the elite gene pool}
        \STATE End;
        \STATE \textbf{return} The results of community detection //{Transforming the most adaptable non-inferior solutions from the elite gene pool into community detection results}
        \end{algorithmic}
        \caption{Framework of F-SGCD Algorithm}
        \label{alg:seq}
    \end{algorithm}
\end{frame}

%------------------------------------------------

%------------------------------------------------
\section{Experimental Results}
%------------------------------------------------

\begin{frame}{Datasets}
    \begin{itemize}
        \item Real-world netowrks\\
        \begin{itemize}
            \item Bottlenose Dolphins
            \item Zachary's Karate Club
            \item Football
            \item Krebs' book
        \end{itemize}
        \item Artificial networks \\
        \begin{itemize}
            \item LFR (Lancichinetti-Fortunato-Radicchi) Benchmark
        \end{itemize}
    \end{itemize}
\end{frame}

%------------------------------------------------

\begin{frame}{Evaluation Metrics}
    \begin{itemize}
        \item Normalized Mutual Information (NMI) \\
        It is used to measure the similarity between the detected communities and the known communities. Given two partitions $A$ and $B$ of a network in communities, let $C$ be the confusion matrix whose element $C_{ij}$ is the number of nodes of community $i$ of the partition $A$ that are also in the community $j$ of the partition $B$.
        \item Modularity \\
        It is a criterion for evaluating the quality of community detection.
    \end{itemize}
\end{frame}

%------------------------------------------------

\begin{frame}{Experiments on synthetic LFR networks}
    \begin{columns}
        % Column 1
        \begin{column}{0.5\textwidth}
                It can be observed that the algorithm performs relatively better than the others
        \end{column}
        % Column 2    
        \begin{column}{0.5\textwidth}
            \begin{figure}
            \centering
                \includegraphics[width=0.8\textwidth]{synthetic-LFR.PNG}
                \caption{Comparison of F-SGCD and other algorithm.}
            \end{figure}
        \end{column}
    \end{columns}
\end{frame}

%------------------------------------------------

\begin{frame}{Experiments on synthetic real-world networks}
    \begin{columns}
        \begin{column}{0.5\textwidth}
            \begin{figure}
            \centering
                \includegraphics[width=1\textwidth]{synthetic-realworld.PNG}
                \caption{Q values of the eight compared algorithms on four real-world networks, averaging
                over 20 runs.}
            \end{figure}
        \end{column}
    \end{columns}
\end{frame}

%------------------------------------------------

\begin{frame}{Network hierarchy of Pareto solution}
    \begin{itemize}
        \item Hierarchal modularity \\
        In a hierarchical modular network, many small-scale vertices with dense internal connections are loosely connected, and thus forming a larger-scale topology module. This topological structure is arranged in hierarchical order, and the network that generates modules iteratively is called hierarchical network.
        \item Bottlenose Dolphins
    \end{itemize}
\end{frame}

%------------------------------------------------

\begin{frame}{References}
    % Beamer does not support BibTeX so references must be inserted manually as below
    \footnotesize{
        \begin{thebibliography}{99}
            \bibitem[Li-Cao-Ding-Li, 2020]{p1} Qi Li, Zehong Cao, Weiping Ding, Qing Li (2020)
            \newblock A multi-objective adaptive evolutionary algorithm to extract communities in networks
            \newblock \emph{Swarm and Evolutionary Computation} 100629.
        \end{thebibliography}
    }
\end{frame}

%------------------------------------------------

\begin{frame}
    \Huge{\centerline{\textbf{The End}}}
\end{frame}

%----------------------------------------------------------------------------------------

\end{document}